nifi-extension-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/PutMongo.java

```diff
@@ -183,16 +204,16 @@ protected Collection<ValidationResult> customValidate(final ValidationContext va
 
     @Override
     public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
-        final FlowFile flowFile = session.get();
+        FlowFile flowFile = session.get();
         if (flowFile == null) {
             return;
         }
 
         final ComponentLog logger = getLogger();
 
         final Charset charset = Charset.forName(context.getProperty(CHARACTER_SET).getValue());
-        final String mode = context.getProperty(MODE).getValue();
-        final String updateMode = context.getProperty(UPDATE_MODE).getValue();
+        final String processorMode = context.getProperty(MODE).getValue();
+        final String flowfileType = context.getProperty(UPDATE_OPERATION_MODE).getValue();
```

Comment:
The `flowfileType` is not descriptive of the property value, so recommend `updateOperationMode`.
```suggestion
        final String updateOperationMode = context.getProperty(UPDATE_OPERATION_MODE).getValue();
```